### ISSUE TITLE
[influxdb-enterprise] - Missing support for LDAP

### ISSUE THREAD

**Issue by pdrastil** [influxdata/helm-charts#229]
I was looking into official Helm chart for InfluxDB Enterprise that is currently missing LDAP authentication. I would like to open the discussion how to implement this functionality because it's one of the features that is required for project I'm working on. This feature should work with StartTLS option and support custom certificate used by LDAP server.

Following are areas that blocks me from creating PR for your repository

1)  Documentation is missing information where to place custom certificates for LDAP or how to configure them. Should they be present in data nodes, meta nodes or both? Is there some config section in TOML that is undocumented for setting path to this certificate or it's assumed that certificate is installed on the OS level? If it's OS level it would mean running `sudo update-ca-certificates` and thus running container as a root on Kubernetes which is a big security issue or baking the certificates into container which means maintenance and upgrade overhead.

2) Documentation is missing detailed information about usage of LDAP TOML config file. Is this file used only once when `influxd-ctl ldap set-config /path/to/ldap.toml` is invoked or is the configuration used every time when container starts and it should be bound as a volume to meta nodes?

3) If configuration TOML is used only for one time setup and data are then stored in the database, how can this be achieved by bootstrap job? From what I've seen `influxd-ctl` doesn't support parameters to target cluster from bootstrap container so only option would be to use `curl` to invoke API. Problem is that this API is undocumented so I don't know how the payload should be formatted.

Thank you for answering this.

**Comment by pdrastil** [comment]
## TL;DR

I've tried to implement this feature and decided to abandon this request because I haven't found reasonable solution that would work reliably. I've tried to add LDAP into bootstrap container as well directly into meta container as a ConfigMap.

## Problems

When LDAP configuration gets added into bootstrap container you cannot upgrade LDAP role bindings or LDAP bind password because boostrap runs only during first deployment of the chart. When configuration is added into meta node as a volume, LDAP integration works and can be changed during helm upgrades. This solution works only when meta node authentication is disabled. This is unacceptable for production use so I've tried to enable this feature. At this point you will get into situation that rest of deployed nodes cannot join the cluster because they have to authenticate with LDAP administrator password. Providing this password in another Secret would solve this, however at this point LDAP integration that you want to use for automatic password rotation becomes pointless.
